PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/localrun/FunctionRunState.java

```diff
@@ -246,18 +262,56 @@ private ProcessBuilder getRunFunctionCliProcessBuilder(File stagingFolder, int f
         return processBuilder;
     }
 
+    private ProcessBuilder getRunFunctionCliExtensionInstallProcessBuilder(File stagingFolder) {
+        final ProcessBuilder processBuilder = new ProcessBuilder();
+        final String funcPath = functionRunConfiguration.getFuncPath();
+        String[] command = new String[]{funcPath, "extensions", "install", "--java"};
+        processBuilder.command(command);
+        processBuilder.directory(stagingFolder);
+        return processBuilder;
+    }
+
     private void prepareStagingFolder(File stagingFolder, RunProcessHandler processHandler) throws AzureExecutionException {
         ReadAction.run(() -> {
             final Path hostJsonPath = FunctionUtils.getDefaultHostJson(project);
             final Path localSettingsJson = Paths.get(functionRunConfiguration.getLocalSettingsJsonPath());
             final PsiMethod[] methods = FunctionUtils.findFunctionsByAnnotation(functionRunConfiguration.getModule());
             try {
-                FunctionUtils.prepareStagingFolder(stagingFolder.toPath(), hostJsonPath, functionRunConfiguration.getModule(), methods);
-                FunctionUtils.copyLocalSettingsToStagingFolder(stagingFolder.toPath(), localSettingsJson, functionRunConfiguration.getAppSettings());
+                Map<String, FunctionConfiguration> configMap =
+                        FunctionUtils.prepareStagingFolder(stagingFolder.toPath(), hostJsonPath, functionRunConfiguration.getModule(), methods);
+                FunctionUtils.copyLocalSettingsToStagingFolder(stagingFolder.toPath(),
+                                                                                localSettingsJson, functionRunConfiguration.getAppSettings());
+
+                final Set<BindingEnum> bindingClasses = getFunctionBindingEnums(configMap);
```

Comment:
We may refactor the function extension install related logic to FunctionUtils later